patcher.sh

```diff
@@ -0,0 +1,330 @@
+	fi
+
+	# Cleanup in case the dirs/files exists
+	rm -rf "$OUTPUT" "$SCRIPT_DIR/AM2R.AppDir" "$SCRIPT_DIR/AM2R-x86_64.AppImage" "$SCRIPT_DIR/AM2RWrapper/" "$SCRIPT_DIR/utilities/android/assets/" "$SCRIPT_DIR/AM2RWrapper.apk"
+
+	# Create necessary directories
+	mkdir -p "$OUTPUT" "$RESOURCES"
+
+	# Check for AM2R_11
+	if [[ -f "$AM2RZIP" ]]; then
+		echo "AM2R_11.zip found!"
+		unzip -q "$AM2RZIP" -d "$OUTPUT"
+	elif [[ -d "$AM2RZIP" ]]; then
+		echo "AM2R_11 folder found!"
+		cp -R "$AM2RZIP" "$OUTPUT"
+	else
+		echo "AM2R_11 not found! Please place AM2R_11.zip (case sensitive) into \"$SCRIPT_DIR\" or provide it via command line arguments and try again."
+		return 1
+	fi
+
+	# Check for which OS we patch
+	if [ "$OSCHOICE" = "linux" ]; then
+
+		# Check whether Xdelta is installed
+		if [ ! command -v xdelta3 &> /dev/null ] ; then
+			>&2 echo "Xdelta is not installed! Please install 'xdelta3' from your local package manager!"
+			return 1
+		fi
+
+		echo "Patching for Linux..."
+		echo "Applying AM2R xdelta patch..."
+		xdelta3 -dfs "$OUTPUT/AM2R.exe" "$SCRIPT_DIR/data/AM2R.xdelta" "$OUTPUT/runner"
+
+		echo "Applying data xdelta patch..."
+		xdelta3 -dfs "$OUTPUT/data.win" "$SCRIPT_DIR/data/game.xdelta" "$RESOURCES/game.unx"
+
+		echo "Cleaning up residual AM2R 1.1 files..."
+		rm "$OUTPUT/AM2R.exe" "$OUTPUT/data.win" "$OUTPUT/D3DX9_43.dll"
+
+		echo "Formatting game directory..."
+		# This won't move the runner, because it doesn't have a dot
+		mv $OUTPUT/*.* "$RESOURCES/"
+
+		echo "Installing new datafiles..."
+		cp -R "$SCRIPT_DIR"/data/files_to_copy/* "$RESOURCES/"
+
+		if [ "$HQMUSIC" = true ]; then
+			echo "Copying HQ music..."
+			cp "$SCRIPT_DIR"/data/HDR_HQ_in-game_music/*.ogg "$RESOURCES/"
+		fi
+
+		# Format music for Unix, aka lowercase it
+		# Finds every file, and if its not already lowercase, lowercases it.
+		find "$RESOURCES" -type f -exec bash -c '
+		    target="{}";
+		    cd "$(dirname "${target}")";
+		    target="$(basename "${target}")";
+		    ! [[ "${target}" = "${target,,}" ]] && mv "${target}" "${target,,}"
+		' \;
+
+		# Remove old lang folder
+		rm -R "$OUTPUT"/lang
+
+		chmod +x "$OUTPUT/runner"
+
+		if [ "$SYSTEMWIDE" = true ]; then
+			mkdir -p "$PREFIX/bin"
+			if [ "$APPIMAGE" = true ]; then
+				ln -sf "$OUTPUT/AM2R.AppImage" "$PREFIX/bin/am2r"
+			else
+				ln -sf "$OUTPUT/runner" "$PREFIX/bin/am2r"
+			fi
+		fi
+
+		# Create .desktop file
+		echo "Creating desktop file..."
+		cp "$SCRIPT_DIR/data/files_to_copy/icon.png" "$OUTPUT/icon.png"
+		local desktopPath="$OUTPUT/AM2R.desktop"
+		if [ "$SYSTEMWIDE" = true ]; then
+			mkdir -p "$PREFIX/share/applications"
+			desktopPath="$PREFIX/share/applications/AM2R.desktop"
+		fi
+		cp "$SCRIPT_DIR/DesktopTemplate" "$desktopPath"
+
+		# replace [REPLACE] with OUTPUT.Replace("/", "\/") in desktop file
+		sed -i "s/\[REPLACE\]/${OUTPUT//\//\\\/}/" "$desktopPath"
+
+		# If we don't want to create an AppImage, we're done.
+		if [ "$APPIMAGE" = false ]; then
```

Comment:
The message saying the operation was succesfull could be moved to the bottom, so that it also gets printed when creating an appimage. Something like:
```suggestion
		if [ "$APPIMAGE" = true ]; then
                    ...
                else
                    sed -i "s/AM2R.AppImage/runner/" "$desktopPath"
                fi
                
               	echo "The operation was completed successfully. See you next mission!"
 ```

Reply:
Didn't think of that. Applied locally.